release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -78,18 +93,26 @@ public static int compareInstances(GKInstance inst1, GKInstance inst2, StringBui
 	 * that will count as 1 diff and the elements will NOT be compared.
 	 */
 	public static int compareInstances(
-		GKInstance inst1, GKInstance inst2, StringBuilder sb, int maxRecursionDepth, boolean checkReferrers
+		GKInstance instance1, GKInstance instance2, StringBuilder stringBuilder,
+		int maxRecursionDepth, boolean checkReferrers
 	)
 	{
-		return compareInstances(inst1, inst2, sb, 0, 0, maxRecursionDepth, null, checkReferrers);
+		final int diffCount = 0;
+		final int initialRecursionDepth = 0;
+		final Predicate<? super SchemaAttribute> customAttributeNameFilter = null;
```

Comment:
Why is this object being created and immediately assigned null? I don't think it actually gets used. Why not just use `null` instead of `customAttributeNameFilter` whose value is null?

Reply:
Same goes for `diffCount` and `initialRecursionDepth`, I guess.

Reply:
I wanted to give those values, even though they are null and 0, names so it is clear what is being passed to the `compareInstances` method on the next line.  It is hard to understand at a glance what the parameter is when the literal value is passed.